### DISCUSSION TITLE
Multiprocessing

### DISCUSSION THREAD

**Discussion by ArtyomMaiorov** [py-pdf/pypdf_table_extraction#8] [Q&A]
Hi! Is multiprocessing available for reading tables in pdf?

**Comment by foarsitter** [comment]
Can you elaborate a little more why you want to use multiprocessing? Currently it isn't supported.

**Comment by ArtyomMaiorov** [comment]
I have a program which scrapes relatively small tables from thousands of pdfs and processes them. I use camelot.read_pdf() to extract them. Currently, the read_pdf() is the main slowing point. I wonder if there is a way to speed this process up.

**Comment by bosd** [comment]
FYI, There was a related WIP in https://github.com/camelot-dev/camelot/pull/237

**Comment by foarsitter** [comment]
multiprocessing can be done in many ways and I'm not convinced that we need to support it out of the box. 

You have a producer (your scraper) and a consumer (camelot), that is a common paradigm. I don't know if you store your pdfs in between but if you do you can start multiple processes consuming the pdf's. Doesn't seems something we need to support out of the box.

**Reply by ArtyomMaiorov** [reply]
I will look into it, thanks!

**Reply by foarsitter** [reply]
This looks promising: https://gist.github.com/ownport/63167dbb162f998964f309a5046bef58

**Comment by bosd** [comment]
This is implemented in https://github.com/py-pdf/pypdf_table_extraction/pull/17
